Homework2/Trie/Trie.cs

```diff
@@ -0,0 +1,219 @@
+﻿namespace Trie;
+using System.Collections;
+
+/// <summary>
+/// Trie-structure.
+/// </summary>
+public class Trie
+{
+    private readonly TrieElement head = new();
+    private TrieElement? triePointer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Trie"/> class.
+    /// </summary>
+    public Trie()
+    {
+        this.triePointer = this.head;
+    }
```

Comment:
```suggestion
    public Trie()
        => this.triePointer = this.head;
```
С конструкторами тоже так можно, хотя обычно не нужно, потому что они часто больше одного поля инициализируют